### DISCUSSION TITLE
Is the M-BUS module needed?

### DISCUSSION THREAD

**Discussion by Kasinge** [UtilitechAS/amsreader-firmware#485] [Q&A] [answered]
In the hardware section of the wiki https://github.com/UtilitechAS/amsreader-firmware/wiki/hardware it is stated that a TSS721 M-BUS module board should be used with an ESP8266 or ESP32. However, I have been running a Wemos D1 mini with the Kamstrup meter directly connected to ground and data out, and power supply via USB following this guide https://smarthjemmet.dk/2021/09/aflaes-kamstrup-elmaalere-med-esp8266-eller-esp32/.

What is the function of the M-bus module, and should I use one? I recently bought a Wemos D32 that I would like to upgrade to, because the UI is a bit slow on the ESP8266 and hopefully will work better with the faster board. Do I need the M-bus module with the D32?

**Comment by gskjold** [comment, marked as answer]
This whole project started for meters in Norway, where a HAN port is made available on all meters, using M-Bus on the hardware layer. For your Kamstrup meter this is not needed when you connect to the 6-pin plug in the module bay, as this is a pure TTL port and not M-Bus.

I guess it would be a good time to update the wiki a bit on this matter, as the connection depends on what meter you have.

**Reply by Kasinge** [reply]
Thank you
